environment/src/main/java/com/sequenceiq/environment/environment/flow/stop/handler/StopFreeIpaHandler.java

```diff
@@ -43,12 +43,16 @@ public void accept(Event<EnvironmentDto> environmentDtoEvent) {
         EnvironmentDto environmentDto = environmentDtoEvent.getData();
         try {
             if (Strings.isNullOrEmpty(environmentDto.getParentEnvironmentCrn())) {
-                freeIpaService.describe(environmentDto.getResourceCrn()).ifPresent(freeIpa -> {
-                    if (freeIpa.getStatus() != null && !freeIpa.getStatus().isStoppable()) {
+                freeIpaService.describe(environmentDto.getResourceCrn()).ifPresentOrElse(freeIpa -> {
+                    if (freeIpa.getStatus() != null && (freeIpa.getStatus().isStoppedPhase() || freeIpa.getStatus().isStopInProgressPhase())) {
+                        LOGGER.info("Freeipa is already {}, so it won't started", freeIpa.getStatus());
+                    } else if (freeIpa.getStatus() != null && !freeIpa.getStatus().isStoppable()) {
```

Comment:
I would move the null check one layer up as it used in two condition paths and in case of stop there is no status for the FreeIPA stack that should be a critical error.

Reply:
I didn't want to add a new level to here